lib/Models/Subscriber.php

```diff
@@ -340,6 +340,9 @@ public function customFields() {
     )->select_expr(MP_SUBSCRIBER_CUSTOM_FIELD_TABLE . '.value');
   }
 
+  /**
+   * @deprecated Will be removed after 2021/09/30. Use MailPoet\Subscribers\SubscriberSaveController
```

Comment:
I think we should also add `trigger_error('Calling ' . $methodName . ' is deprecated and will be removed.', E_USER_DEPRECATED);` so that everyone still using this will get a notice.